doc/1.7/dev_guide/lua_style_guide.rst

```diff
@@ -0,0 +1,465 @@
+-------------------------------------------------------------------------------
+                                Lua Style Guide
+-------------------------------------------------------------------------------
+
+Inspiration:
+
+* https://github.com/Olivine-Labs/lua-style-guide
+* http://dev.minetest.net/Lua_code_style_guidelines
+* http://sputnik.freewisdom.org/en/Coding_Standard
+
+Programming style is an art. There is some arbitrariness to the rules, but there
+are sound rationales for them. It is useful not only to provide sound advice on
+style but to understand the underlying rationale and human aspect of why the
+style recommendations are formed:
+
+* http://mindprod.com/jgloss/unmain.html
+* http://www.oreilly.com/catalog/perlbp/
+* http://books.google.com/books?id=QnghAQAAIAAJ
+
+Zen of Python is good; understand it and use wisely:
+
+    | Beautiful is better than ugly.
+    | Explicit is better than implicit.
+    | Simple is better than complex.
+    | Complex is better than complicated.
+    | Flat is better than nested.
+    | Sparse is better than dense.
+    | Readability counts.
+    | Special cases aren't special enough to break the rules.
+    | Although practicality beats purity.
+    | Errors should never pass silently.
+    | Unless explicitly silenced.
+    | In the face of ambiguity, refuse the temptation to guess.
+    | There should be one -- and preferably only one -- obvious way to do it.
+    | Although that way may not be obvious at first unless you're Dutch.
+    | Now is better than never.
+    | Although never is often better than *right* now.
+    | If the implementation is hard to explain, it's a bad idea.
+    | If the implementation is easy to explain, it may be a good idea.
+    | Namespaces are one honking great idea -- let's do more of those!
+    |
+    | https://www.python.org/dev/peps/pep-0020/
+
+===========================================================
+                 Indentation and Formatting
+===========================================================
+
+* 4 spaces instead tabs. PIL suggests using of two spaces, but programmer looks
+  at code 4 up to 8 hours a day, so it's simplier to distinguish indentation
+  with 4 spaces. Why spaces? Similar representation everywhere.
+
+  You can use vim modelines:
+
+  .. code-block:: lua
+
+      -- vim:ts=4 ss=4 sw=4 expandtab
+
+* No whitespaces at EOL, newline at end of file (?)
```

Comment:
Wording is a little confusing here. An file should ends w/ one newline symbol, but shouldn't ends w/ blank line (two newline symbols).